src/utils/era5_observations_to_forcing_file.jl

```diff
@@ -0,0 +1,419 @@
+# This file converts ERA5 observational data to forcing data for ClimaAtmos
+# single column model runs. For some terms we use the ERA5 data directly,
+# including air temperature, specific humidity, and wind. For advective tendencies
+# we need to compute gradient terms in the vertical and horizontal which is
+# computed in separate helper functions. ERA5 data is available from ECMWF
+# split over different files for variables available at levels, surface variables
+# measured instantaneously (surface temperature), and variables that are reported
+#as accumulated (sensible and latent heat fluxes), which must be divided by
+# the accumulation period (typically 1 hour or 1 day in seconds).
+
+using NCDatasets
+using Statistics
+using Dates
+
+# for radiation calculation
+import Insolation
+import Insolation.Parameters as IP
+import ClimaParams as CP
+
+"""
+    get_external_forcing_file_path(parsed_args; data_dir)
+
+Get the path to the external forcing file for a given site and start date.
+When using the BUILDKITE env, a temporary directory is used for the
+external forcing file. Otherwise, the file is expected to stored in the
+era5_hourly_atmos_processed artifact directory.
+"""
+function get_external_forcing_file_path(
+    parsed_args;
+    data_dir = get(ENV, "BUILDKITE", "") == "true" ? mktempdir() :
+               @clima_artifact("era5_hourly_atmos_processed"),
+)
+    start_date = parsed_args["start_date"]
+    # round to era5 quarter degree resolution for site selection
+    site_latitude = round(parsed_args["site_latitude"] * 4) / 4
+    site_longitude = round(parsed_args["site_longitude"] * 4) / 4
+
+    if site_latitude != parsed_args["site_latitude"] ||
+       site_longitude != parsed_args["site_longitude"]
+        @info "Rounded site latitude/longitude from ($(parsed_args["site_latitude"]), $(parsed_args["site_longitude"])) to ($(site_latitude), $(site_longitude)) for ERA5 quarter-degree resolution."
+    end
+
+    return joinpath(
+        data_dir,
+        "tv_forcing_$(site_latitude)_$(site_longitude)_$(start_date).nc",
+    )
+end
+
+"""
+    check_external_forcing_file_times(forcing_file_path, parsed_args)
+
+Check that the simulation start and end times are within the range of the external forcing file.
+"""
+function check_external_forcing_file_times(forcing_file_path, parsed_args)
+    start_date = parsed_args["start_date"]
+    start_time = Dates.DateTime(start_date, "yyyymmdd")
+    t_end = parsed_args["t_end"]
+    end_date = start_time + Dates.Second(time_to_seconds(t_end))
+    NCDataset(forcing_file_path) do ds
+        @assert ds["time"][1] <= start_time "Start time $start_time is before the first time step in the forcing file"
+        @assert ds["time"][end] >= end_date "End time $end_date is after the last time step in the forcing file"
+    end
+    return true
+end
```

Comment:
What we typically refer to `time` is in units of `s`, not at date. 

```suggestion
function check_external_forcing_file_times(forcing_file_path, parsed_args)
    start_date = Dates.DateTime(parsed_ards["start_date"], "yyyymmdd")
    t_end = parsed_args["t_end"]
    end_date = start_date + Dates.Second(time_to_seconds(t_end))
    NCDataset(forcing_file_path) do ds
        @assert ds["time"][1] <= start_date "Start time $start_date is before the first time step in the forcing file"
        @assert ds["time"][end] >= end_date "End time $end_date is after the last time step in the forcing file"
    end
    return true
end
```